core/controllers/acl_decorators.py

```diff
@@ -45,6 +47,24 @@
 (suggestion_models,) = models.Registry.import_models([models.NAMES.suggestion])
 
 
+def handle_redirection(handler, redirection_url, expected_return_type):
```

Comment:
Also, try not to start function names with "handle". It is vague. Have them "do" something -- e.g. redirect(), redirect_permanently(), etc.